event-sourcing/event-source-api-integration-test/src/test/java/uk/gov/justice/services/eventsourcing/source/api/EventsFeedIT.java

```diff
@@ -0,0 +1,377 @@
+import org.apache.openejb.testing.Classes;
+import org.apache.openejb.testing.Configuration;
+import org.apache.openejb.testing.EnableServices;
+import org.apache.openejb.testing.Module;
+import org.apache.openejb.testng.PropertiesBuilder;
+import org.apache.openejb.util.NetworkUtil;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@EnableServices("jaxrs")
+@RunWith(ApplicationComposer.class)
+public class EventsFeedIT {
+    private static final String NAME = "Test Name";
+    private static final String PAYLOAD_JSON = "{\"field\": \"Value\"}";
+    private static final String METADATA_JSON = "{\"field\": \"Value\"}";
+    private final static ZonedDateTime TIMESTAMP = new UtcClock().now();
+
+
+    private static final String LIQUIBASE_EVENT_STORE_CHANGELOG_XML = "liquibase/event-store-db-changelog.xml";
+    private static final String BASE_URI_PATTERN = "http://localhost:%d/event-source-api/rest";
+    private static final UUID SYSTEM_USER_ID = randomUUID();
+    private static int port = -1;
+
+    private CloseableHttpClient httpClient;
+    @Resource(name = "openejb/Resource/eventStore")
+    private DataSource dataSource;
+
+    @Inject
+    private OpenEjbAwareEventRepository eventsRepository;
+
+    @Inject
+    private TestEventsFeedService eventFeedService;
+
+
+    @BeforeClass
+    public static void beforeClass() {
+        port = NetworkUtil.getNextAvailablePort();
+    }
+
+    @Before
+    public void setup() throws Exception {
+        httpClient = HttpClients.createDefault();
+        initEventDatabase();
+    }
+
+    @Configuration
+    public Properties properties() {
+        return new PropertiesBuilder()
+                .p("httpejbd.port", Integer.toString(port))
+                .build();
+    }
+
+    @Module
+    @Classes(cdi = true, value = {
+            EventFeedResource.class,
+            OpenEjbAwareEventRepository.class,
+            TestEventsFeedService.class,
+            AccessController.class,
+            EventsFeedIT.TestSystemUserProvider.class,
+            ForbiddenRequestExceptionMapper.class,
+            LoggerProducer.class,
+            TestEventInsertionStrategyProducer.class
+
+    })
+    public WebApp war() {
+        return new WebApp()
+                .contextRoot("event-source-api")
+                .addServlet("TestApp", Application.class.getName())
+                .addInitParam("TestApp", "javax.ws.rs.Application", EventSourceApiApplication.class.getName());
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        eventFeedService
+                .initialiseWithPageSize(25);
+    }
+
+    @Test
+    public void shouldReturnFirstPageOfFeed() throws Exception {
+
+        final UUID streamId = randomUUID();
+
+        eventFeedService.initialiseWithPageSize(3);
+
+        eventsRepository.insert(eventOf(1L, streamId));
+        eventsRepository.insert(eventOf(2L, streamId));
```

Comment:
it be good to explicitly setup the json payload here, so when you assert against it, it's clear when it's coming from,
also it'd be good to insert events with different payloads